api/proxy/store/builder/config.go

```diff
@@ -17,34 +17,41 @@ import (
 	"github.com/urfave/cli/v2"
 )
 
-// Config ... Higher order config which bundles all configs for building
-// the proxy store manager with necessary client context
+// Config ... Higher order storage layer config which bundles
+// all subconfigs for building the proxy storage backends and their
+// respective clients
 type Config struct {
+	// main storage storage enablement config that dictates
+	// secondary backend policies and EigenDA backends
 	StoreConfig store.Config
 
-	// main storage configs
+	KzgConfig kzg.KzgConfig
+
 	ClientConfigV1   common.ClientConfigV1
 	VerifierConfigV1 verify.Config
-	KzgConfig        kzg.KzgConfig
-	ClientConfigV2   common.ClientConfigV2
+
+	ClientConfigV2 common.ClientConfigV2
 
 	MemstoreConfig  *memconfig.SafeConfig
 	MemstoreEnabled bool
 
-	// secondary storage cfgs
+	// secondary storage cfg
 	S3Config s3.Config
 }
 
 // ReadConfig ... parses the Config from the provided flags or environment variables.
+// client config ingestion is ignored if `memstore=true`
 func ReadConfig(ctx *cli.Context) (Config, error) {
+	useMemStore := ctx.Bool(memstore.EnabledFlagName)
```

Comment:
you say that "client config ingestion is ignored if memstore=true", but thats not really true because you populate 2 fields, which I'm guessing are read somewhere else to populate some other config or whatnot.

Would prefer if we take a bit more time to actually split those things apart completely. We just just not even have to call client.ReadConfig at all when useMemStore is set to true. Like this should just return an empty config right here